rust/nft-candy-machine/src/lib.rs

```diff
@@ -447,7 +461,17 @@ pub struct AddConfigLines<'info> {
     #[account(signer)]
     authority: AccountInfo<'info>,
 }
+#[derive(Accounts)]
+pub struct WithdrawFunds<'info> {
+    #[account(
+        mut,
+        has_one = authority
+    )]
+    config: ProgramAccount<'info, Config>,
 
+    #[account(signer, address = config.authority)]
+    authority: AccountInfo<'info>,
```

Comment:
Also authority should be a Signer, right now its not, which means anybody can empty this.

Reply:
I believed that this line

 #[account(signer, address = config.authority)]

ensured authority both needs to be signer and needs to be the authority in config.authority?

I did try with a non-authority address and it didn't let me have access

@bhgames

Reply:
That's deprecated. Use Signer.